src/SFML/Graphics/ImageLoader.cpp

```diff
@@ -61,6 +61,15 @@ namespace
         sf::InputStream* stream = static_cast<sf::InputStream*>(user);
         return stream->tell() >= stream->getSize();
     }
+
+    // stb_image callback for constructing a buffer
+    // This assumes 'context' is an std::vector<sf::Uint8>*
+    void bufferFromCallback(void* context, void* data, int size)
+    {
+        sf::Uint8* ptr = &static_cast<sf::Uint8*>(data)[0];
+        sf::Uint8* end = ptr + size;
+        static_cast<std::vector<sf::Uint8>*>(context)->assign(ptr, end);
```

Comment:
`assign`? Are you sure? It would be rather inefficient for stb_image to have a storage on its side and then provide all the data at once, rather than calling the callback whenever it has new data to write. But I assume that you have tested your code and that it works fine.

Reply:
I got caught up in STB's implementation for png and didn't realize it did actually callback more than once for other formats. The fix will be commited soon.